velox/functions/prestosql/DecimalFunctions.cpp

```diff
@@ -203,6 +203,43 @@ struct DecimalDivideFunction {
   uint8_t aRescale_;
 };
 
+template <typename TExec>
+struct DecimalModulusFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(TExec);
+
+  template <typename A, typename B>
+  void initialize(
+      const std::vector<TypePtr>& inputTypes,
+      const core::QueryConfig& /*config*/,
+      A* /*a*/,
+      B* /*b*/) {
+    auto aType = inputTypes[0];
+    auto bType = inputTypes[1];
+    auto [aPrecision, aScale] = getDecimalPrecisionScale(*aType);
+    auto [bPrecision, bScale] = getDecimalPrecisionScale(*bType);
+    aRescale_ = computeRescaleFactor(aScale, bScale);
+    bRescale_ = computeRescaleFactor(bScale, aScale);
+  }
+
+  template <typename R, typename A, typename B>
+  void call(R& out, const A& a, const B& b) {
+    R quotient;
+    out = DecimalUtil::divide<R, A, B>(
```

Comment:
Its possible this code has changed now but did you intend to use divideWithRoundUp ?